src/java.base/share/classes/jdk/internal/misc/ThreadTracker.java

```diff
@@ -39,15 +39,25 @@ public class ThreadTracker {
      * The hashCode/equals methods do not invoke the Thread hashCode/equals method
      * as they may run arbitrary code and/or leak references to Thread objects.
      */
-    private record ThreadRef(Thread thread) {
+    private class ThreadRef {
+        private final Thread thread;
+
+        ThreadRef(Thread thread) {
+            this.thread = thread;
+        }
+
+        Thread thread() {
+            return this.thread;
+        }
+
         @Override
         public int hashCode() {
-            return Long.hashCode(thread.threadId());
+            return Long.hashCode(thread.getId());
         }
```

Comment:
`threadId()` method does not exit in Java 11 yet
we are using `getId()` method which is doing exactly the same job.
- https://github.com/openjdk/jdk/blame/4018b2b19629ddb8cd7a56e064dfef371f23e5fa/src/java.base/share/classes/java/lang/Thread.java#L2565

```java
    @Deprecated(since="19")
    public long getId() {
        return threadId();
    }

    public final long threadId() {
        return tid;
    }
```